proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResource.java

```diff
@@ -63,24 +66,27 @@ public View handlePostBinding(
             @FormParam(SamlFormMessageType.SAML_REQUEST) String encodedEidasAuthnRequest,
             @FormParam(RelayState.DEFAULT_ELEMENT_LOCAL_NAME) String eidasRelayState,
             @Session HttpSession session) {
```

Comment:
Can we use the `relayState` supplied by connector as the redis key for our data? This is provided in the AuthnRequest and returned as the `relayState` in the Hub's response to Proxy Node.
In this way we can remove any dependency on `@Session HttpSession session` which will make the gateway stateless, avoiding a set of bugs around session management.

Reply:
Sounds good 🙂 lets stick this in a new ticket and discuss it with the team. @JHjava what do you think?

Reply:
Good point Phil and yes, if we can be sure RelayState will always be populated. Currently the eidasRelayState in GatewaySessionData isn't specified as NotEmpty. If not RelayState then something else instead of sessionId. We should look into this. I agree with Tom, lets stick this in a different ticket.